opentech/apply/funds/forms.py

```diff
@@ -69,28 +67,42 @@ def __init__(self, *args, **kwargs):
         self.request = kwargs.pop('request', None)
         super().__init__(*args, **kwargs)
 
-        reviewers = self.instance.reviewers.all()
         self.submitted_reviewers = User.objects.filter(id__in=self.instance.reviews.values('author'))
 
         if self.can_alter_external_reviewers(self.instance, self.user):
-            self.prepare_field('reviewer_reviewers', reviewers, self.submitted_reviewers)
+            reviewers = self.instance.reviewers.all()
+            self.prepare_field(
+                'reviewer_reviewers',
+                initial=reviewers,
+                excluded=self.submitted_reviewers
+            )
         else:
             self.fields.pop('reviewer_reviewers')
 
+        self.roles = {}
         staff_reviewers = User.objects.staff()
+
+        assigned_roles = {
+            assigned.role: assigned.reviewer
+            for assigned in self.instance.assigned.filter(
+                role__isnull=False
+            )
+        }
+
         for role in ReviewerRole.objects.all().order_by('order'):
-            field_name = 'reviewer_' + str(role.pk)
+            field_name = 'reviewer_role_' + str(role)
```

Comment:
The reason I made this role.pk and not role.name is because the name could have spaces or any sort of other characters that don't work for form fields names. I think it's easiest to use pk, but I guess you could also do something fancy in the __str__ method on the model.

Reply:
I can see your point. I was just trying to make it a bit more meaningful. 

I guess the options are [slugify](https://docs.djangoproject.com/en/2.1/ref/utils/#django.utils.text.slugify) it or use the id.